cmake/add_builtin.cmake

```diff
@@ -47,9 +47,46 @@ function(add_builtin)
     message(STATUS "Adding builtin ${DESCRIPTION}")
 
     add_library(${LIB_NAME} STATIC ${SRC})
-    target_link_libraries(${LIB_NAME} PRIVATE spidermonkey extension_api ${LINK_LIBS})
+
+    if (DEPENDENCIES)
+        message(VERBOSE "${LIB_NAME} depends on ${DEPENDENCIES}")
+
+        add_dependencies(${LIB_NAME} ${DEPENDENCIES})
+
+        foreach(DEPENDENCY IN ITEMS ${DEPENDENCIES})
+            get_target_property(TYPE ${DEPENDENCY} TYPE)
+            if(NOT ${TYPE} MATCHES "UTILITY")
+                # A built-in can either depend on libraries implicitly
+                # linked by StarlingMonkey (i.e. depending on OpenSSL
+                # for your built-in when it only needs OpenSSL::Crypto),
+                # or on a library target.
```

Comment:
I think/hope that OpenSSL is actually not linked in if it's not depended on by any other target.

Reply:
Done!